commandline.c

```diff
@@ -33,38 +36,67 @@ static void usage(const char *prgname)
 
 int main(int argc, char* argv[])
 {
+	extern char *optarg;
+	extern int optind;
+	bool print_help = false, print_version = false;
+	int c = 0;
 	char *iface = NULL;
 	int ret = -1;
 	bool load_embedded = true;
 	const char *progname = "ap51-flash";
 
-	if (argc >= 1)
-		progname = argv[0];
+	while ((c = getopt(argc, argv, "hvm:")) != -1) {
+		switch (c) {
+			case 'h':
+				print_help = true;
+				ret = 0;
+				break;
+			case 'v':
+				print_version = true;
+				ret = 0;
+				break;
+			case 'm':
+				mac_whitelist_add(optarg);
+				break;
+			default:
+				print_help = true;
```

Comment:
please always add a break. If you don't add one (because it should fall-through) at a fall through comment like this:

    case FOO:
    	a = 1;
    	/* fall-through */
    case BAR:
    	b = 2;
    	break;